test/Chainweb/Test/Pact/DbCacheTest.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Chainweb.Test.Pact.DbCacheTest (tests) where
+
+import Chainweb.Pact.Backend.DbCache
+
+import Control.Monad.State.Strict
+
+import Data.Aeson
+import Data.ByteString.Lazy (toStrict)
+
+import Database.SQLite3.Direct
+
+import GHC.Compact
+
+import Test.Tasty
+import Test.Tasty.HUnit
+
+tests :: TestTree
+tests = testGroup "Chainweb.Test.Pact.DbCacheTest"
+    [ testCache ]
+
+defaultChunkSize :: Int
+defaultChunkSize = 32768
+
+-- | Create entries with a size the corresponds to the default
+-- chunk size of the cache.
```

Comment:
There's a lot of internals here that are exposed to the test making it difficult to understand.

1) What is the origin of `defaultChunkSize`

2) Why do we care about chunk size when creating test data? Modules are arbitrarily large/small so it seems like we're crafting the test to the implementation here.

Reply:
1. Origin of default chunk size is in `ghc-compact`. It's an implementation constant for compact regions.

2. We care about the chunk size in order to test cache eviction. The the size of cache entries depends on this value. The size of the input is crafted so that it triggers eviction. The precise values don't matter, just the rough order.

We are crafting the test inputs such that the cache exhibits the behavior that we want to test.